gcloud-java-storage/src/main/java/com/google/gcloud/spi/DefaultStorageRpc.java

```diff
@@ -465,15 +465,18 @@ public void write(String uploadId, byte[] toWrite, int toWriteOffset, long destO
       GenericUrl url = new GenericUrl(uploadId);
       HttpRequest httpRequest = storage.getRequestFactory().buildPutRequest(url,
           new ByteArrayContent(null, toWrite, toWriteOffset, length));
-      long limit = destOffset + length;
-      StringBuilder range = new StringBuilder("bytes ");
-      range.append(destOffset).append('-').append(limit - 1).append('/');
-      if (last) {
-        range.append(limit);
-      } else {
-        range.append('*');
+      // 0-length requests are valid, don't need content-range.
```

Comment:
The 0-length write happens when we call Files.createFile, I hadn't tried to write after that. I added a test for this, and it worked fine. This isn't quite the same as getting an outputstream to write 0 bytes and then more; the build-in buffering gets in the way of testing that.